PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,478 @@
+
+  TString paramName = "L1prefiring_muonparam_0.0To0.2_" + dataeraMuon_;
+  parametrization0p0To0p2_ = (TF1*)file_prefiringparams_->Get(paramName);
+  paramName = "L1prefiring_muonparam_0.2To0.3_" + dataeraMuon_;
+  parametrization0p2To0p3_ = (TF1*)file_prefiringparams_->Get(paramName);
+  paramName = "L1prefiring_muonparam_0.3To0.55_" + dataeraMuon_;
+  parametrization0p3To0p55_ = (TF1*)file_prefiringparams_->Get(paramName);
+  paramName = "L1prefiring_muonparam_0.55To0.83_" + dataeraMuon_;
+  parametrization0p55To0p83_ = (TF1*)file_prefiringparams_->Get(paramName);
+  paramName = "L1prefiring_muonparam_0.83To1.24_" + dataeraMuon_;
+  parametrization0p83To1p24_ = (TF1*)file_prefiringparams_->Get(paramName);
+  paramName = "L1prefiring_muonparam_1.24To1.4_" + dataeraMuon_;
+  parametrization1p24To1p4_ = (TF1*)file_prefiringparams_->Get(paramName);
+  paramName = "L1prefiring_muonparam_1.4To1.6_" + dataeraMuon_;
+  parametrization1p4To1p6_ = (TF1*)file_prefiringparams_->Get(paramName);
+  paramName = "L1prefiring_muonparam_1.6To1.8_" + dataeraMuon_;
+  parametrization1p6To1p8_ = (TF1*)file_prefiringparams_->Get(paramName);
+  paramName = "L1prefiring_muonparam_1.8To2.1_" + dataeraMuon_;
+  parametrization1p8To2p1_ = (TF1*)file_prefiringparams_->Get(paramName);
+  paramName = "L1prefiring_muonparam_2.1To2.25_" + dataeraMuon_;
+  parametrization2p1To2p25_ = (TF1*)file_prefiringparams_->Get(paramName);
+  paramName = "L1prefiring_muonparam_2.25To2.4_" + dataeraMuon_;
+  parametrization2p25To2p4_ = (TF1*)file_prefiringparams_->Get(paramName);
+  paramName = "L1prefiring_muonparam_HotSpot_" + dataeraMuon_;
+  parametrizationHotSpot_ = (TF1*)file_prefiringparams_->Get(paramName);
+
+  produces<double>("nonPrefiringProb").setBranchAlias("nonPrefiringProb");
+  produces<double>("nonPrefiringProbUp").setBranchAlias("nonPrefiringProbUp");
+  produces<double>("nonPrefiringProbDown").setBranchAlias("nonPrefiringProbDown");
+
+  produces<double>("nonPrefiringProbJet").setBranchAlias("nonPrefiringProbJet");
+  produces<double>("nonPrefiringProbJetUp").setBranchAlias("nonPrefiringProbJetUp");
+  produces<double>("nonPrefiringProbJetDown").setBranchAlias("nonPrefiringProbJetDown");
+
+  produces<double>("nonPrefiringProbPhoton").setBranchAlias("nonPrefiringProbPhoton");
+  produces<double>("nonPrefiringProbPhotonUp").setBranchAlias("nonPrefiringProbPhotonUp");
+  produces<double>("nonPrefiringProbPhotonDown").setBranchAlias("nonPrefiringProbPhotonDown");
+
+  produces<double>("nonPrefiringProbMuon").setBranchAlias("nonPrefiringProbMuon");
+  produces<double>("nonPrefiringProbMuonUp").setBranchAlias("nonPrefiringProbMuonUp");
+  produces<double>("nonPrefiringProbMuonDown").setBranchAlias("nonPrefiringProbMuonDown");
+}
+
+L1PrefiringWeightProducer::~L1PrefiringWeightProducer() {
+  delete h_prefmap_photon;
+  delete h_prefmap_jet;
+  delete parametrization0p0To0p2_;
```

Comment:
Are the TF1s owned by the TFile? If so, then this would be a double delete.

Reply:
I have added a `Close()` for that file, as it is done for the file for the ECAL prefiring maps. I think that relinquishes the ownership and these deletes make sense.

Reply:
I'm fairly certain `Close()` doesn't relinquish the memory.

Reply:
Ok, I have removed these deletes.